src/js/BookReader.js

```diff
@@ -2985,13 +3041,36 @@ BookReader.prototype.fragmentFromParams = function(params) {
   }
 
   // search
-  if (params.search) {
+  if (params.search && urlMode === 'hash') {
     fragments.push('search', params.search);
   }
 
   return utils.encodeURIComponentPlus(fragments.join(separator)).replace(/%2F/g, '/');
 };
 
+/**
+ * Create, update querystring from the params object
+ *
+ * @param {Object} params
+ * @param {string} currQueryString
+ * @param {string} [urlMode]
+ * @return {string}
+ */
+BookReader.prototype.queryStringFromParams = function(
```

Comment:
Question:  This new core code seems to exclusively munge the URL - can we put this in the URL Plugin?

Reply:
We'll look at this later